common/metrics/otel_metrics_handler.go

```diff
@@ -79,13 +79,13 @@ func (omp *otelMetricsHandler) Counter(counter string) CounterIface {
 
 // Gauge obtains a gauge for the given name and MetricOptions.
 func (omp *otelMetricsHandler) Gauge(gauge string) GaugeIface {
-	c, err := omp.provider.GetMeter().AsyncFloat64().Gauge(gauge)
+	c, err := omp.provider.GetMeter().SyncFloat64().UpDownCounter(gauge)
```

Comment:
@yiminc I am not sure about this. You fix this by using AsyncFloat64. This failed the test after upgrade.